### ISSUE TITLE
SO(3) radius querries

### ISSUE THREAD

**Issue by barcode** [jlblancoc/nanoflann#112]
I have a set of orientations and need to find all orientations in a radius.
In my case, the radius is the smallest delta orientation around any axis (equivalent to `Eigen::Quaternionf::angularDistance`).

When looking at the `nanoflann::SO3_Adaptor`, I saw it is using `nanoflann::L2_Simple_Adaptor` and therefore would generate incorrect results for my metric.

My question:
Is it possible to use a metric equivalent to `Eigen::Quaternionf::angularDistance and what would I have to do for this?

**Comment by jlblancoc** [comment]
You are right, the non-Euclidean part of the library needs an clean up and upgrade...
